libraries/AP_IBus_Telem/AP_IBus_Telem.cpp

```diff
@@ -0,0 +1,507 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+   i-BUS telemetry for FlySky/Turnigy receivers and other peripherals
+   (eg iA6B, iA10) by Nicole Ashley <nicole@ashley.kiwi>.
+
+   Originally based on work by Jan Verhulst:
+     https://github.com/ArduPilot/ardupilot/pull/16545
+
+   Libraries used for reference and inspiration:
+
+   * iBUStelemetry
+     https://github.com/Hrastovc/iBUStelemetry
+
+   * IBusBM
+     https://github.com/bmellink/IBusBM
+
+   * BetaFlight
+     https://github.com/betaflight/betaflight/blob/master/src/main/telemetry/ibus_shared.c
+ */
+
+#include <AP_IBus_Telem/AP_IBus_Telem.h>
+
+#if HAL_IBUS_TELEM_ENABLED
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Vehicle/AP_Vehicle.h>
+#include <AP_BattMonitor/AP_BattMonitor.h>
+#include <AP_Arming/AP_Arming.h>
+#include <AP_GPS/AP_GPS.h>
+#include <AP_RPM/AP_RPM.h>
+
+void AP_IBus_Telem::init()
+{
+    const AP_SerialManager &serial_manager = AP::serialmanager();
+    if ((_port = serial_manager.find_serial(AP_SerialManager::SerialProtocol_IBUS_Telem, 0))) {
+        _port->set_options(_port->OPTION_HDPLEX);
+        hal.scheduler->register_timer_process(FUNCTOR_BIND_MEMBER(&AP_IBus_Telem::tick, void));
+    }
+}
+
+void AP_IBus_Telem::tick(void)
+{
+    ensure_listening();
+
+    uint16_t available = MIN(_port->available(), 1024U);
+    if (available == 0) {
+        return;
+    } else {
```

Comment:
```suggestion
    }
```
No `else` clauses after a return statement...